src/debug/HyprCtl.cpp

```diff
@@ -13,95 +13,386 @@
-std::string layersRequest() {
+std::string layersRequest(bool json) {
     std::string result = "";
 
-    for (auto& mon : g_pCompositor->m_vMonitors) {
-        result += getFormat("Monitor %s:\n", mon->szName.c_str());
-        int layerLevel = 0;
-        for (auto& level : mon->m_aLayerSurfaceLists) {
-            result += getFormat("\tLayer level %i:\n", layerLevel);
-
-            for (auto& layer : level) {
-                result += getFormat("\t\tLayer %x: xywh: %i %i %i %i, namespace: %s\n", layer, layer->geometry.x, layer->geometry.y, layer->geometry.width, layer->geometry.height, layer->szNamespace.c_str());
+    if (json) {
+        result += "{\n";
+
+        for (auto& mon : g_pCompositor->m_vMonitors) {
+            result += getFormat(
+R"#("%s": {
+    "levels": {
+)#",
+                mon->szName.c_str()
+            ); 
+
+            int layerLevel = 0;
+            for (auto& level : mon->m_aLayerSurfaceLists) {
+                result += getFormat(
+R"#(        
+        "%i": [
+)#",
+                    layerLevel
+                );
+                for (auto& layer : level) {
+                    result += getFormat(
+R"#(                {
+                    "address": "0x%x",
+                    "x": %i,
+                    "y": %i,
+                    "w": %i,
+                    "h": %i,
+                    "namespace": "%s"
+                },)#",
+					layer,
+                    layer->geometry.x,
+                    layer->geometry.y,
+                    layer->geometry.width,
+                    layer->geometry.height,
+                    layer->szNamespace.c_str()
+                    );
+                }
+
+                // remove trailing comma
+                result.pop_back();
+
+                if (level.size() > 0)
+                    result += "\n        ";
+
+                result += "],";
+
+                layerLevel++;
             }
 
-            layerLevel++;
+            // remove trailing comma
+            result.pop_back();
+
+            result += "\n    }\n},";
+        }
+
+        // remove trailing comma
+        result.pop_back();
+
+        result += "\n}\n";
+        
+    } else {
+        for (auto& mon : g_pCompositor->m_vMonitors) {
+            result += getFormat("Monitor %s:\n", mon->szName.c_str());
+            int layerLevel = 0;
+            for (auto& level : mon->m_aLayerSurfaceLists) {
+                result += getFormat("\tLayer level %i:\n", layerLevel);
+
+                for (auto& layer : level) {
+                    result += getFormat("\t\tLayer %x: xywh: %i %i %i %i, namespace: %s\n", layer, layer->geometry.x, layer->geometry.y, layer->geometry.width, layer->geometry.height, layer->szNamespace.c_str());
+                }
+
+                layerLevel++;
+            }
+            result += "\n\n";
         }
-        result += "\n\n";
     }
 
     return result;
 }
 
-std::string devicesRequest() {
+std::string devicesRequest(bool json) {
```

Comment:
to every single with bool json.

NO.

Make an enum with modes. Name it for example eHyprctlOutputFormat

and make it
FORMAT_NORMAL = 0,
FORMAT_JSON

then pass that.